dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/SdkFeature.kt

```diff
@@ -166,6 +170,39 @@ internal class SdkFeature(
 
     // endregion
 
+    // region Context Update Listener
+
+    internal fun notifyContextUpdated(context: Map<String, Any?>) {
+        contextUpdateListeners.keys.forEach {
+            it.onContextUpdate(context)
+        }
+    }
+
+    internal fun setContextUpdateListener(listener: FeatureContextUpdateListener) {
+        synchronized(contextUpdateListeners) {
+            // the argument is always non - null, so we can suppress the warning
+            @Suppress("UnsafeThirdPartyFunctionCall")
+            if (contextUpdateListeners.contains(listener)) {
+                internalLogger.log(
+                    InternalLogger.Level.WARN,
+                    InternalLogger.Target.USER,
+                    { CONTEXT_UPDATE_LISTENER_ALREADY_EXISTS.format(Locale.US, wrappedFeature.name) }
```

Comment:
But we don't actually overwrite anything, do we? Both key and value stay the same.

Reply:
yes you are right...unless you are providing a different instance of the same listener. The `equals` method will say true and we are replacing the previous instance with the new one.